ui/js/controllers/perf/alerts.js

```diff
@@ -110,6 +111,41 @@ perf.controller('AlertsCtrl', [
                 });
         };
 
+        function updateAlertVisibility() {
+            _.forEach($scope.alertSummaries, function(alertSummary) {
+                _.forEach(alertSummary.alerts, function(alert) {
+                    // only show alert if it passes all filter criteria
+                    alert.visible =
+                        (alert.series_signature.framework_id === $scope.filterOptions.framework.id) &&
+                        (!$scope.filterOptions.hideImprovements || alert.is_regression) &&
+                        _.every($scope.filterOptions.filter.split(' '),
+                                function(matchText) {
+                                    return !matchText ||
```

Comment:
If there's no matchText, we shouldn't filter on it. So we check for its absence.